src/main/java/org/ow2/proactive/catalog/service/BucketService.java

```diff
@@ -188,8 +188,12 @@ public BucketMetadata getBucketMetadata(long id) {
 
     public List<BucketMetadata> listBuckets(String ownerName, String kind) {
         List<BucketEntity> entities;
+        List<String> owners = new ArrayList<>();
+        owners.add(ownerName);
 
-        if (!StringUtils.isEmpty(ownerName)) {
+        if (!StringUtils.isEmpty(ownerName) && !StringUtils.isEmpty(kind)) {
+            entities = bucketRepository.findByOwnerIsInContainingKind(owners, kind);
```

Comment:
Thanks for useful comments.
In the end of the day the secure method listBuckets(List<String> owners, String kind) and insecure listBuckets(String ownerName, String kind) method will call the same DB method, but only in case when the owner and kind is not empty. It will be better for system tests.
I didn't change the behavior of insecure mode (just added check when both parameters are present), before it was just ignoring kind when the both kind and owner are presented.
This is clear shown in the issue: https://github.com/ow2-proactive/catalog/issues/57